packages/runtime/container-runtime/src/summaryFormat.ts

```diff
@@ -93,11 +93,19 @@ export type GCVersion = number;
 export interface IGCMetadata {
     /**
      * The version of the GC code that was run to generate the GC data that is written in the summary.
+     * If the persisted value doesn't match the current value in the code, saved GC data will be discarded and regenerated from scratch.
      * Also, used to determine whether GC is enabled for this container or not:
      * - A value of 0 or undefined means GC is disabled.
      * - A value greater than 0 means GC is enabled.
      */
     readonly gcFeature?: GCVersion;
+    /**
+     * The "GC Container Generation" is provided by the container author to denote GC-impacting changes to the container's structure.
+     * The value provided at creation time is persisted, and compared with the value provided at each load.
+     * If the persisted value doesn't match the value provided on load, GC Sweep will be disabled to protect against dataloss in case
+     * containers with an older Generation were found to have GC-impacting bugs.
+     */
+    readonly gcContainerGeneration?: number;
```

Comment:
What's the difference between this and shipping a patch that changes the feature flag change? For example, if we find a bug in int3.0.x  and we have a fix in int3.1.x we can just ship a patch that changes the feature flag version. Or if sweep works for versions int3.0.x and int3.2.x but we shipped a regression in int3.1.x, we can always patch the new feature flag name to int3.0.x and int3.2.x and turn off the old feature flag name and we would get even better granularity.

Reply:
This decouples from our versions and puts control in the hands of the container devs (Loop, WB, etc).  That is exactly what _they_ would do to bump this - check in a bump to the value they're passing to us and deploy it.

Reply:
Presumably they'd check in the bump with or after a fix to a GC-blocking bug.

Reply:
I see, so if we ship a bug, we would use potentially feature flags, or something like that. If they ship a bug, they would pass in `gcContainerGeneration`

Reply:
(unresolving for visibility)

Reply:
We should probably try to articulate all the classes of bugs we can think of and see if we have mitigations in place to "move on" from them.

The two that are on my mind now:
* FF bug resulting in incorrect GC Data (but actual references are correct)
  * We bump the GC Version (aka gcFeature)
  * GC will discard the bad data and regenerate it going forward.
* A bug in how the container or a component stores references between objects
  * The container dev bumps gcContainerGeneration
  * Any docs created with the code with the bug will no longer be eligible for Sweep (and its benefits...)

We should think if there are other cases.

Reply:
The other advantage is this will resolve issues faster as they don't need us to release a patch and then integrate it which can be really expensive and time consuming at times.